test/comps/test_rf.cc

```diff
@@ -0,0 +1,163 @@
+#include "RandLAPACK.hh"
+#include "rl_blaspp.hh"
+#include "rl_lapackpp.hh"
+
+#include <RandBLAS.hh>
+
+#include <fstream>
+#include <gtest/gtest.h>
+
+class TestRF : public ::testing::Test
+{
+    protected:
+
+    virtual void SetUp() {};
+
+    virtual void TearDown() {};
+
+    template <typename T>
+    struct RFTestData {
+        std::vector<T> A;
+        std::vector<T> A_cpy;
+        std::vector<T> Q;
+        std::vector<T> Buf1;
+        std::vector<T> Buf2;
+        std::vector<T> Q_cpy;
+        std::vector<T> Q_hat_cpy;
+
+        RFTestData(int64_t m, int64_t n, int64_t k) :
+        A(m * n, 0.0), 
+        A_cpy(m * n, 0.0),
+        Q(m * k, 0.0), 
+        Buf1(m * m, 0.0), 
+        Buf2(m * m, 0.0), 
+        Q_cpy(m * k, 0.0), 
+        Q_hat_cpy(m * n, 0.0) 
+        {}
+    };
+
+    template <typename T, typename RNG>
+    static void computational_helper(int64_t m, int64_t n, RFTestData<T>& all_data) {
+        
+        lapack::lacpy(MatrixType::General, m, n, all_data.A.data(), m, all_data.A_cpy.data(), m);
+        
+        RandLAPACK::CholQRQ<T> CholQRQ(false, false);
+        CholQRQ.call(m, n, all_data.A_cpy);
+
+        lapack::lacpy(MatrixType::General, m, n, all_data.A_cpy.data(), m, all_data.Q_hat_cpy.data(), m);
+    }
+
+    /// General test for QB:
+    /// Computes QB factorzation, and checks:
+    /// 1. A - QB
+    /// 2. B - \transpose{Q}A
+    /// 3. I - \transpose{Q}Q
+    /// 4. A_k - QB = U_k\Sigma_k\transpose{V_k} - QB
+    template <typename T, typename RNG>
+    static void test_RF_general(int64_t m, int64_t n, int64_t k, int64_t p, RandBLAS::base::RNGState<RNG> state, RFTestData<T>& all_data) {
+
+        printf("|==================================TEST QB2 GENERAL BEGIN==================================|\n");
+
+        //Subroutine parameters
+        bool verbosity = false;
+        bool cond_check = true;
+        int64_t passes_per_iteration = 1;
+
+        // Make subroutine objects
+        // Stabilization Constructor - Choose PLU
+        RandLAPACK::PLUL<T> Stab(cond_check, verbosity);
+        // RowSketcher constructor - Choose default (rs1)
+        RandLAPACK::RS<T, RNG> RS(Stab, state, p, passes_per_iteration, verbosity, cond_check);
+        // Orthogonalization Constructor - Choose CholQR
+        RandLAPACK::HQRQ<T> Orth_RF(cond_check, verbosity);
+        // RangeFinder constructor - Choose default (rf1)
+        RandLAPACK::RF<T> RF(RS, Orth_RF, verbosity, cond_check);
```

Comment:
The RangeFinder algorithm object should be defined in the ``TEST_F`` code blocks, not in this function. This function should accept a ``RangeFinder`` as an argument (passed by address).